text/0009-onramp-offramp-exec.md

```diff
@@ -0,0 +1,77 @@
+- Feature Name: `rfc_0009_onramp_offramp_exec`
+- Start Date: 2020-02-21
+- Rust Issue: [wayfair-incubator/tremor-rfcs#0000](https://github.com/wayfair-incubator/tremor-rfcs/issues/0000)
+- RFC PR: [wayfair-incubator/tremor-rfcs#0000](https://github.com/wayfair-incubator/tremor-rfcs/pull/0000)
+
+# Summary
+[summary]: #summary
+
+Ability to use a script/executable as source of data in onramp. Ability to use a
+script/executable in offramp.
+
+# Motivation
+[motivation]: #motivation
+
+Integrating with various sources of data mandates development of an onramp. Such
+an activity is costly from point of view of development time, Tremor codebase
+growth and maintenance. It would be far easier to just have Tremor run
+scripts/executables in an onramp that would then interface with whatever data
+source necessary. That is not to say that building dedicated onramps/offramps
+does not make sense.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
```

Comment:
as discussed in the meeting earlier today: let's define how stdout and stderr output from the exec commands are treated by the onramp. Could be useful to forward them as distinct outputs from onramp, such that users can connect them differently for their pipeline needs.

Also, would be good to set the command executed as part of the event's origin uri. Users can access that metadata from tremor-script and use it as needed: https://docs.tremor.rs/tremor-script/functions/origin/.